server/src/main/java/org/elasticsearch/action/admin/indices/mapping/put/TransportPutMappingAction.java

```diff
@@ -106,29 +106,36 @@ protected void masterOperation(Task task, final PutMappingRequest request, final
                 listener.onFailure(maybeValidationException.get());
                 return;
             }
-            PutMappingClusterStateUpdateRequest updateRequest = new PutMappingClusterStateUpdateRequest(request.source())
-                .indices(concreteIndices)
-                .ackTimeout(request.timeout()).masterNodeTimeout(request.masterNodeTimeout());
-
-            metadataMappingService.putMapping(updateRequest, new ActionListener<ClusterStateUpdateResponse>() {
-
-                @Override
-                public void onResponse(ClusterStateUpdateResponse response) {
-                    listener.onResponse(new AcknowledgedResponse(response.isAcknowledged()));
-                }
-
-                @Override
-                public void onFailure(Exception t) {
-                    logger.debug(() -> new ParameterizedMessage("failed to put mappings on indices [{}]",
-                        Arrays.asList(concreteIndices)), t);
-                    listener.onFailure(t);
-                }
-            });
+            performMappingUpdate(concreteIndices, request, listener, metadataMappingService);
         } catch (IndexNotFoundException ex) {
             logger.debug(() -> new ParameterizedMessage("failed to put mappings on indices [{}]",
                 Arrays.asList(request.indices())), ex);
             throw ex;
         }
     }
 
+    static void performMappingUpdate(Index[] concreteIndices,
```

Comment:
Extracted the common logic that both regular put mapping api calls and a dynamic mapping update reuse.